src/test/java/com/thirdcc/webapp/web/rest/TransactionResourceIT.java

```diff
@@ -245,100 +456,98 @@ public void getAllTransactionByEventId_WithCancelledEvent_ShouldThrow400() throw
-//
-//        // If the entity doesn't have an ID, it will throw BadRequestAlertException
-//        restTransactionMockMvc.perform(put("/api/transactions")
-//            .contentType(TestUtil.APPLICATION_JSON_UTF8)
-//            .content(TestUtil.convertObjectToJsonBytes(transactionDTO)))
-//            .andExpect(status().isBadRequest());
-//
-//        // Validate the Transaction in the database
-//        List<Transaction> transactionList = transactionRepository.findAll();
-//        assertThat(transactionList).hasSize(databaseSizeBeforeUpdate);
-//    }
-//
-//    @Test
-//    public void deleteTransaction() throws Exception {
-//        // Initialize the database
-//        transactionRepository.saveAndFlush(transaction);
-//
-//        int databaseSizeBeforeDelete = transactionRepository.findAll().size();
-//
-//        // Delete the transaction
-//        restTransactionMockMvc.perform(delete("/api/transactions/{id}", transaction.getId())
-//            .accept(TestUtil.APPLICATION_JSON_UTF8))
-//            .andExpect(status().isNoContent());
-//
-//        // Validate the database contains one less item
-//        List<Transaction> transactionList = transactionRepository.findAll();
-//        assertThat(transactionList).hasSize(databaseSizeBeforeDelete - 1);
-//    }
+
+    @Test
+    public void updateTransaction() throws Exception {
+        // Initialize the database
+        Event savedEvent = initEventDB();
+        transaction.setEventId(savedEvent.getId());
+        transaction.setStatus(DEFAULT_TRANSACTION_STATUS);
+        Transaction savedTransaction = initTransactionDB();
+
+        int databaseSizeBeforeUpdate = transactionRepository.findAll().size();
+
+        // Update the transaction
+        Transaction updatedTransaction = transactionRepository.findById(savedTransaction.getId()).get();
+        // Disconnect from session so that the updates on updatedTransaction are not directly saved in db
+        em.detach(updatedTransaction);
+        updatedTransaction
+            .type(UPDATED_TYPE)
+            .amount(UPDATED_AMOUNT)
+            .details(UPDATED_DETAILS)
+            .status(TransactionStatus.CANCELLED);
+        TransactionDTO transactionDTO = transactionMapper.toDto(updatedTransaction);
+
+        restTransactionMockMvc.perform(put("/api/transactions")
+            .contentType(TestUtil.APPLICATION_JSON_UTF8)
+            .content(TestUtil.convertObjectToJsonBytes(transactionDTO)))
+            .andExpect(status().isOk());
+
+        // Validate the Transaction in the database
+        List<Transaction> transactionList = transactionRepository.findAll();
+        assertThat(transactionList).hasSize(databaseSizeBeforeUpdate);
+        Transaction testTransaction = transactionList.get(transactionList.size() - 1);
+        assertThat(testTransaction.getEventId()).isEqualTo(transaction.getEventId());
+        assertThat(testTransaction.getType()).isEqualTo(transaction.getType());
```

Comment:
same thing. false positive may occur

Reply:
Resolved in 4104de2d993ff69fd961084fe0c4b0424b4afd30